plugins/com.gwtplugins.gwt.eclipse.wtp/src/com/google/gwt/eclipse/wtp/projects/WtpFacetCreatorParicipant.java

```diff
@@ -19,14 +19,14 @@
 
 public class WtpFacetCreatorParicipant implements IWebAppProjectCreator.Participant {
 
-  private IJavaProject javaProject;
+  private IJavaProject[] javaProjects;
   private IFacetedProjectListener projectFacetListener;
  
   @Override
   public void updateWebAppProjectCreator(IWebAppProjectCreator webAppProjectCreator) {
     // Just in case exit early
-    javaProject = webAppProjectCreator.getCreatedJavaProject();
-    if (javaProject == null) {
+    javaProjects = webAppProjectCreator.getCreatedJavaProjects();
+    if (javaProjects == null) {
```

Comment:
it doesnt appear that this should never return null - would it make more sense to test if empty? also, what do you think about `List<IJavaProject>`?

Reply:
changed this to list